src/main/java/org/cactoos/io/LoggingInputStream.java

```diff
@@ -0,0 +1,236 @@
+        if (byts > 0) {
+            this.bytes.getAndAdd(byts);
+            this.time.getAndAdd(millis);
+            if (!level.equals(Level.INFO)) {
+                this.logger.log(
+                    level,
+                    String.format(
+                        msg,
+                        this.bytes.get(),
+                        this.source,
+                        this.time.get()
+                    )
+                );
+            }
+        } else {
+            if (level.equals(Level.INFO)) {
+                this.logger.info(
+                    String.format(
+                        msg,
+                        this.bytes.get(),
+                        this.source,
+                        this.time.get()
+                    )
+                );
+            }
+        }
+        return byts;
+    }
+
+    @Override
+    public long skip(final long num) throws IOException {
+        final long skipped = this.origin.skip(num);
+        this.logger.log(
+            this.logger.getLevel(),
+            String.format(
+                "Skipped %d byte(s) from %s.",
+                skipped,
+                this.source
+            )
+        );
+        return skipped;
+    }
+
+    @Override
+    public int available() throws IOException {
+        final int avail = this.origin.available();
+        this.logger.log(
+            this.logger.getLevel(),
+            String.format(
+                "There is(are) %d byte(s) avaliable from %s.",
+                avail,
+                this.source
+            )
+        );
+        return avail;
+    }
+
+    @Override
+    public void close() throws IOException {
+        this.origin.close();
+        this.logger.log(
+            this.logger.getLevel(),
+            String.format(
+                "Closed input stream from %s.",
+                this.source
+            )
+        );
+    }
+
+    @Override
+    public void mark(final int limit) {
+        this.origin.mark(limit);
+        this.logger.log(
+            this.logger.getLevel(),
+            String.format(
+                "Marked position %d from %s.",
+                limit,
+                this.source
+            )
+        );
+    }
+
+    @Override
+    public void reset() throws IOException {
+        this.origin.reset();
+        this.logger.log(
+            this.logger.getLevel(),
+            String.format(
+                "Reseted input stream from %s.",
+                this.source
+            )
+        );
+    }
+
+    @Override
+    public boolean markSupported() {
+        final boolean supported = this.origin.markSupported();
+        if (supported) {
+            this.logger.log(
+                this.logger.getLevel(),
+                String.format(
+                    "Mark and reset methods are supported from %s.",
+                    this.source
+                )
+            );
+        } else {
+            this.logger.log(
```

Comment:
@fabriciofx when calling ``this.logger.log(this.logger.getLevel(),...)``, isn't it redundant to pass its own level as an argument? Doesn't it have an overloaded method with no LEVEL argument, that uses the encapsulated one? Do you know what I mean?

Reply:
@amihaiemil Yeah, I got it. I'll change it. Thanks!